### PR TITLE
Added diagnostic for @FacesValidator classes missing Validator interface implementation

### LINKED ISSUE #773: Validator classes must implement jakarta.faces.validator.Validator
#### Description:
Introduce a diagnostic rule that enforces validator classes to implement the jakarta.faces.validator.Validator interface. This ensures spec‑compliant usage and prevents runtime errors when custom validators are registered in Jakarta Faces.

#### Specification:
https://jakarta.ee/specifications/faces/3.0/jakarta-faces-3.0#a1414

#### Type of language feature proposed:
_Select all that apply_
- [x] diagnostic
- [ ] quick-fix
- [ ] snippet
- [ ] other, please specify: 

#### Example Code (Non‑Compliant): 

```
import jakarta.faces.validator.FacesValidator; 

@FacesValidator("customValidator") 
public class CustomValidator { 
    // Missing Validator interface implementation 
} 
```

#### Example Code (Compliant): 

```
import jakarta.faces.validator.FacesValidator; 
import jakarta.faces.validator.Validator; 
import jakarta.faces.validator.ValidatorException; 
import jakarta.faces.component.UIComponent; 
import jakarta.faces.context.FacesContext; 

@FacesValidator("customValidator") 
public class CustomValidator implements Validator { 

    @Override 
    public void validate(FacesContext context, UIComponent component, Object value) throws ValidatorException { 
        // validation logic 
    } 
} 
```

### PR DESCRIPTION
**Pull request by sajeerzeji**
Fixes #773 

Implements a diagnostic that detects classes annotated with @FacesValidator that do not implement jakarta.faces.validator.Validator.

https://github.com/user-attachments/assets/288fadfc-195a-4a2d-91c0-efad6adb63cb

### CODE REVIEW

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesValidatorDiagnosticsParticipant.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.eclipse.lsp4jakarta.jdt.core.utils.IJDTUtils;
+import org.eclipse.lsp4jakarta.jdt.core.utils.PositionUtils;
+import org.eclipse.lsp4jakarta.jdt.internal.DiagnosticUtils;
+import org.eclipse.lsp4jakarta.jdt.internal.Messages;
+import org.eclipse.lsp4jakarta.jdt.internal.core.ls.JDTUtilsLSImpl;
+
+/**
+ * {@code @FacesValidator} annotation diagnostic participant.
+ */
+public class FacesValidatorDiagnosticsParticipant implements IJavaDiagnosticsParticipant {
```
Comment (Joseph-Bineesh) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesValidatorDiagnosticsParticipant.java]: I would recommend a class name FacesDiagnosticsParticipant

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/resources/org/eclipse/lsp4jakarta/jdt/core/messages/messages.properties [hunk 2]
```diff
@@ -270,3 +270,6 @@ InvalidMultipleInterceptorMethodsOfSameType = Only one method with {0} annotatio
 # SecurityDiagnosticsParticipant - Identity Store Definitions
 MissingApplicationScopedOnIdentityStoreDefinition = A class annotated with @{0} must be annotated with @ApplicationScoped.
 InvalidScopeOnIdentityStoreDefinition = A class annotated with @{0} must be annotated with @ApplicationScoped, instead of @{1}.
+
+# FacesValidatorDiagnosticsParticipant
+FacesValidatorMustImplement = Annotated classes with @FacesValidator must implement the jakarta.faces.validator.Validator interface.
```
Comment (Joseph-Bineesh) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/resources/org/eclipse/lsp4jakarta/jdt/core/messages/messages.properties]: Can we rephrase to 
'Classes annotated with @FacesValidator must implement the jakarta.faces.validator.Validator interface.'

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        IJavaProject javaProject = loadJavaProject("jakarta-sample", "");
+        IFile javaFile = javaProject.getProject().getFile(
+                                                          new Path("src/main/java/io/openliberty/sample/jakarta/faces/DontImplementValidator.java"));
+        String uri = javaFile.getLocation().toFile().toURI().toString();
+
+        JakartaJavaDiagnosticsParams diagnosticsParams = new JakartaJavaDiagnosticsParams();
+        diagnosticsParams.setUris(Arrays.asList(uri));
+
+        // expected: diagnostic on class name (line 5, col 13..35)
+        Diagnostic d = d(5, 13, 35,
```
Comment (Joseph-Bineesh) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java]: Use meaningful name

**Review by Joseph-Bineesh — CHANGES_REQUESTED** [review]

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesDiagnosticsParticipant.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        ICompilationUnit unit = utils.resolveCompilationUnit(uri);
+        List<Diagnostic> diagnostics = new ArrayList<>();
+
+        if (unit == null) {
+            return diagnostics;
+        }
+
+        IType[] alltypes = unit.getAllTypes();
+        for (IType type : alltypes) {
+            IAnnotation[] allAnnotations = type.getAnnotations();
```
Comment (Rejoice472) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesDiagnosticsParticipant.java:55]: Do we need this variable since it is used only once?

Comment (sajeerzeji) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesDiagnosticsParticipant.java:55]: I have refactored this

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java (hunk 3 above)
Comment (sajeerzeji) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java]: Renamed

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesValidatorDiagnosticsParticipant.java (hunk 1 above)
Comment (sajeerzeji) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/faces/FacesValidatorDiagnosticsParticipant.java]: Renamed

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/resources/org/eclipse/lsp4jakarta/jdt/core/messages/messages.properties (hunk 2 above)
Comment (sajeerzeji) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/resources/org/eclipse/lsp4jakarta/jdt/core/messages/messages.properties]: Updated the message, in both lsp4jakarta and LTI

**Review by Rejoice472 — APPROVED** [review]

**Review by Joseph-Bineesh — APPROVED** [review]
LGTM

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4jakarta.commons.JakartaJavaDiagnosticsParams;
+import org.eclipse.lsp4jakarta.jdt.core.utils.IJDTUtils;
+import org.eclipse.lsp4jakarta.jdt.internal.core.ls.JDTUtilsLSImpl;
+import org.eclipse.lsp4jakarta.jdt.test.core.BaseJakartaTest;
+import org.junit.Test;
+
+public class JakartaFacesTest extends BaseJakartaTest {
```
Comment (venmanyarun) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java:32]: Consider edge cases

- A class annotated with @FacesValidator that extends a superclass which implements Validator (inherited implementation) — is doesImplementInterfaces recursive?
- A @FacesValidator applied to an abstract class — should it be exempt?
- A class implementing a raw Validator (without the generic type)

**Review by venmanyarun — CHANGES_REQUESTED** [review]

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java (hunk 5 above)
Comment (sajeerzeji) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/test/faces/JakartaFacesTest.java:32]: From this, the first case was missing to handle.
1. **"A class annotated with @FacesValidator that extends a superclass which implements Validator":** This is handled and tests added.
2. **"A @FacesValidator applied to an abstract class":** This was already been handled, tests added now.
3. **"A class implementing a raw Validator":** This was also already been handled, tests added now.

**Review by venmanyarun — APPROVED** [review]
